File: leetcode/easy/1588. Sum of All Odd Length Subarrays/solution.cc
// https://leetcode.com/problems/sum-of-all-odd-length-subarrays/
#include <iostream>
#include <vector>
#include <numeric>

using namespace std;

class Solution {
public:
    int sumOddLengthSubarrays(vector<int>& arr) {
        int sum = 0;
        for (int i = 1; i <= arr.size(); i += 2) {
            for (int j = 0; i + j <= arr.size(); j += 1) {
                sum = accumulate(arr.begin() + j, arr.begin() + j + i, sum);
            }
        }
        cout << sum;
        return sum;
    }
};

int main() {
    Solution solution;
    vector<int> vec = {1, 4, 2, 5, 3};
    solution.sumOddLengthSubarrays(vec);
    return 0;
}

File: leetcode/easy/1588. Sum of All Odd Length Subarrays/solution.js
// https://leetcode.com/problems/sum-of-all-odd-length-subarrays/
/**
 * @param {number[]} arr
 * @return {number}
 */
var sumOddLengthSubarrays = function(arr) {
    let sum = 0;
    for (let i = 1; i <= arr.length; i += 2) {
        for (let j = 0; j + i <= arr.length; j += 1) {
            sum += arr.slice(j, j+i).reduce((acc, cur) => acc + cur);
        }
    }
    console.log(sum)
    return sum;
};

sumOddLengthSubarrays([1, 4, 2, 5, 3])